cores/esp8266/core_esp8266_waveform.cpp

```diff
@@ -178,17 +178,15 @@ int ICACHE_RAM_ATTR stopWaveform(uint8_t pin) {
   }
   // If user sends in a pin >16 but <32, this will always point to a 0 bit
   // If they send >=32, then the shift will result in 0 and it will also return false
-  uint32_t mask = 1<<pin;
-  if (!(waveformEnabled & mask)) {
-    return false; // It's not running, nothing to do here
-  }
-  waveformToDisable |= mask;
-  // Ensure timely service....
-  if (T1L > microsecondsToClockCycles(10)) {
-    timer1_write(microsecondsToClockCycles(10));
-  }
-  while (waveformToDisable) {
-    /* no-op */ // Can't delay() since stopWaveform may be called from an IRQ
+  if (waveformsEnabled & (1UL << pin)) {
+    waveformToDisable = 1UL << pin;
```

Comment:
Everyone can read 1UL << pin, absolutely no need to introduce an object mask for this idiom. Use constness if you really feel the need to.

Reply:
Removing a named variable like this, used immediately after definition in the code, is not going to affect GCC's generated code unless it's god a really bad hangover. It's for readability by the next maintainer, only.